servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectLBHttpConnectionFactory.java

```diff
@@ -161,4 +175,23 @@ private static void removeH1Handlers(final Channel channel) {
             pipeline.remove(handlerClass);
         }
     }
+
+    private static Throwable handleException(final Throwable cause, final Channel channel) {
+        // If the Channel is still active, close it in case of any error to free resources.
+        if (channel.isActive()) {
+            channel.close();
```

Comment:
should the `ChannelFuture` be consumed and logged in case of something failed during close?

Reply:
Looking at how we close the channel in any other place we never do that. The primary `cause` is the most important exception that we should propagate. Closing the channel is the best effort. I never saw a case when it can fail. Did you?
But while we are here, we should use `ChannelCloseUtils.close(channel, cause)` instead of a regular `channel.close()`.

Reply:
IIRC I have only seen this in the past if the shutdown procedure races the channel closes (i.e. someone closes the event loops beforehand and then the channel complains at some point that it is shut down). While this would help identify an illegal usage / race condition - thinking about it again in this use case we likely won't benefit from it at all. So let's go with `ChannelCloseUtils` 👍